promql/lex.go

```diff
@@ -198,21 +198,20 @@ var key = map[string]itemType{
 	"stdvar": itemStdvar,
 
 	// Keywords.
-	"alert":         itemAlert,
-	"if":            itemIf,
-	"for":           itemFor,
-	"labels":        itemLabels,
-	"annotations":   itemAnnotations,
-	"offset":        itemOffset,
-	"by":            itemBy,
-	"without":       itemWithout,
-	"keeping_extra": itemKeepCommon,
-	"keep_common":   itemKeepCommon,
-	"on":            itemOn,
-	"ignoring":      itemIgnoring,
-	"group_left":    itemGroupLeft,
-	"group_right":   itemGroupRight,
-	"bool":          itemBool,
```

Comment:
As discussed in off-PR, let's move the removed keyword into the "Removed keywords" section at the bottom here (and the same in the reverse map!), plus add an if condition in `nextItem()` to print a helpful warning message if someone uses it.